test/extended/util/framework.go

```diff
@@ -2054,6 +2056,25 @@ func SkipIfExternalControlplaneTopology(oc *CLI, reason string) {
 	}
 }
 
+type DoesApiResourceExistFn func(apiResourceName string) bool
```

Comment:
I have introduce this function to limit the number of resource discovery calls and support multiple invocations in a single test